Sitka-Sound-Science-Center/sitka-landslide
# Add google analytics and fix typo

## Overview

The main story here is adding the Google Analytics snippet.  Since this is a Next.js app, I wasn't sure exactly where it should go and how it would need to be wrapped, and a bit of research led me to the conclusion that I should use a package.
The Next.js maintainers have exactly the thing, [`@next/third-parties`](https://www.npmjs.com/package/@next/third-parties), but it requires newer versions of Node and Next.js.  Fortunately there's a very similar package, [`nextjs-google-analytics`](https://www.npmjs.com/package/nextjs-google-analytics), that works with older versions.

Resolves #91

In addition to that, this PR contains a few other small fixes:
- Removing the extra "s" in the "Partners" list on the "About" page (issue #92)
- Removing the `package-lock.json` file. It presumably got added early on before we settled on the `yarn`-based build process, and was never used.  And `yarn` would sometimes show warnings about it being there.
- Adding `--no-use` to the line in `scripts/activate_yvm` where it sources `nvm.sh`.  That line would sometimes fail with no error message, so the script that was calling it would just exit without having done anything.  This was one solution suggested on threads about the problem, and it seems to make it work consistently.

### Notes

I don't have a way of validating that the analytics data is actually getting sent.  I can see that it loads `https://www.googletagmanager.com/gtag/js?id=NONE` ("NONE" is the fake ID), but even if I started it up with the real ID in development, I wouldn't be able to see the dashboard to know if it's actually getting data.  My inclination is to say that it's probably fine and we should just run with it then come back and look closer if it turns out it's not.

## Testing Instructions

As noted just above, I don't think we're in a position to test the Google Analytics fully, but you can...
- Run `./scripts/update` to get the new package and `./scripts/server` to start the dev server.
- Open http://localhost:3008/ and confirm in the network tab that it makes a request to googletagmanager.com with the fake ID.

## Checklist
